packages/astro/src/core/config/merge.ts

```diff
@@ -36,9 +36,7 @@ function mergeConfigRecursively(
 				};
 				continue;
 			}
-		}
-
-		if (key === 'data' && rootPath === 'db') {
+		} else if (key === 'data' && rootPath === 'db') {
```

Comment:
Can this be reverted? I think it'd be more consistent with the rest of the code here.

Reply:
This code block contains top-level if conditions to do certain things, using `else if` means that it's related to the previous `if` statement, which it isn't.

I don't think it's worth changing this and the performance gain isn't a lot.

Reply:
Ah， I get you